spec/pane-spec.coffee

```diff
@@ -153,6 +153,89 @@ describe "Pane", ->
+        itemD = new Item("D")
+        pane.activateItem(itemC, pending: true)
+
+        expect(itemC in pane.getItems()).toBe true
+        expect(pane.getActiveItem()).toBe itemC
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(pane.isActiveItemPending()).toBe true
+
+        pane.activateItem(itemD, pending: true)
+
+        expect(itemC in pane.getItems()).toBe false
+        expect(pane.getActiveItem()).toBe itemD
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(pane.isActiveItemPending()).toBe true
+
+        pane.activateItem(itemD, pending: true)
+
+        expect(pane.getActiveItem()).toBe itemD
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(pane.isActiveItemPending()).toBe true
+
+        pane.activateItem(itemB)
+
+        expect(itemD in pane.getItems()).toBe false
+        expect(pane.getActiveItem()).toBe itemB
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(pane.isActiveItemPending()).toBe false
+
+    fdescribe "when pending item is re-activated", ->
+      it "invokes ::onDidConfirmPendingItem() observers if `pending` is not true", ->
+        itemC = new Item("C")
+        events = []
+        pane.onDidConfirmPendingItem (event) -> events.push(event)
+        pane.activateItem(itemC, pending: true)
+        pane.activateItem(itemC)
+        expect(itemC in pane.getItems()).toBe true
+        expect(pane.getActiveItem()).toBe itemC
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(events).toEqual [{item: itemC, index: 1}]
+        expect(pane.isActiveItemPending()).toBe false
+
+      it "does not invoke ::onDidConfirmPendingItem() observers if `pending` is true", ->
+        itemD = new Item("D")
+        events = []
+        pane.onDidConfirmPendingItem (event) -> events.push(event)
+        pane.activateItem(itemD, pending: true)
+        pane.activateItem(itemD, pending: true)
+        expect(itemD in pane.getItems()).toBe true
+        expect(pane.getActiveItem()).toBe itemD
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(events).toEqual []
+        expect(pane.isActiveItemPending()).toBe true
+
+    fdescribe "when pending item's buffer content is changed", ->
+      it "invokes ::onDidConfirmPendingItem() observers", ->
+        itemC = new Item("C")
+        events = []
+        pane.onDidConfirmPendingItem (event) -> events.push(event)
+        pane.activateItem(itemC, pending: true)
+        console.log 'active item', pane.getActiveItem()
+        console.log '####', pane.getActiveItem() instanceof TextEditor
+        expect(itemC in pane.getItems()).toBe true
+        expect(pane.getActiveItem()).toBe itemC
+        expect(pane.getActiveItemIndex()).toBe 1
+        expect(events).toEqual [{item: itemC, index: 1}]
+        expect(pane.isActiveItemPending()).toBe false
+
+
+    fdescribe "when pending item is saved or destroyed", ->
+      it "invokes ::onDidConfirmPendingItem() observers", ->
```

Comment:
Hmm, not sure it makes sense to confirm the pending item if it's destroyed... what's your thinking behind doing it if the item is saved. Seems like to be saved it will have to be modified, which should confirm it instead.